testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractIntegrationSpec.groovy

```diff
@@ -195,22 +195,102 @@ abstract class AbstractIntegrationSpec extends Specification {
+     * <p>
+     * Use {@link #buildScript(java.lang.String)} to <b>set (replace)</b> the entire file contents instead.
+     */
+    void buildFile(TestFile buildFile, @GroovyBuildScriptLanguage String append) {
+        buildFile << append
+    }
+
+    /**
+     * <b>Appends</b> provided code to the {@link #getSettingsFile() default settings file}.
+     * <p>
+     * Use {@link #settingsScript(java.lang.String)} to <b>set (replace)</b> the entire file contents instead.
+     */
+    void settingsFile(@GroovySettingsScriptLanguage String append) {
+        settingsFile << append
+    }
+
+    /**
+     * <b>Appends</b> provided code to the given settings file.
+     * <p>
+     * Use {@link #settingsScript(java.lang.String)} to <b>set (replace)</b> the entire file contents instead.
+     */
+    void settingsFile(String settingsFile, @GroovySettingsScriptLanguage String append) {
+        file(settingsFile) << append
+    }
+
+    /**
+     * <b>Appends</b> provided code to the given settings file.
+     * <p>
+     * Use {@link #settingsScript(java.lang.String)} to <b>set (replace)</b> the entire file contents instead.
+     */
+    void settingsFile(TestFile settingsFile, @GroovySettingsScriptLanguage String append) {
+        settingsFile << append
+    }
+
+    /**
+     * <b>Appends</b> provided code to the {@link #getInitScriptFile() default init script file}.
+     * <p>
+     * Use {@link #initScript(java.lang.String)} to <b>set (replace)</b> the entire file contents instead.
+     */
+    void initScriptFile(@GroovyInitScriptLanguage String append) {
+        initScriptFile << append
     }
 
-    void settingsFile(@GroovyBuildScriptLanguage String script) {
-        groovyFile(settingsFile, script)
+    /**
+     * <b>Appends</b> provided code to the given init script file.
+     * <p>
+     * Use {@link #initScript(java.lang.String)} to <b>set (replace)</b> the entire file contents instead.
+     */
+    void initScriptFile(String initScriptFile, @GroovyInitScriptLanguage String append) {
+        file(initScriptFile) << append
+    }
+
+    /**
+     * <b>Appends</b> provided code to the given init script file.
+     * <p>
+     * Use {@link #initScript(java.lang.String)} to <b>set (replace)</b> the entire file contents instead.
+     */
+    void initScriptFile(TestFile initScriptFile, @GroovyInitScriptLanguage String append) {
+        initScriptFile << append
     }
 
     /**
      * Provides best-effort groovy script syntax highlighting.
      * The highlighting is imperfect since {@link GroovyBuildScriptLanguage} uses stub methods to create a simulated script target environment.
+     *
+     * @deprecated Use a more specific method, e.g {@link #buildFile(java.lang.String, java.lang.String)} or {@link #settingsFile(java.lang.String, java.lang.String)}
      */
+    @Deprecated
     void groovyFile(String targetBuildFile, @GroovyBuildScriptLanguage String script) {
         groovyFile(file(targetBuildFile), script)
     }
```

Comment:
❗: This method according my research is not that used (7 uses)
Would you consider instead of a deprecation just replacing the use-cases?

Reply:
Good idea! Done ✔️ 

Though, in the end, I modified the `groovyFile` methods to provide just-Groovy syntax highlight. This could be useful when file content is required for Groovy sources (e.g. in `buildSrc/src/main/groovy`)

Reply:
I can see the utility there, thanks!